sdks/rust/src/sdk.rs

```diff
@@ -12,349 +12,211 @@
+    errors::{Result},
+};
+
+#[inline]
+fn empty() -> api::Empty {
+    api::Empty {}
+}
 
 /// SDK is an instance of the Agones SDK
+#[derive(Clone)]
 pub struct Sdk {
-    client: Arc<sdk_grpc::SdkClient>,
-    health: Arc<Mutex<Option<grpcio::ClientCStreamSender<sdk::Empty>>>>,
-    health_receiver: Arc<Mutex<Option<grpcio::ClientCStreamReceiver<sdk::Empty>>>>,
-    alpha: Arc<Alpha>,
+    client: SdkClient<Channel>,
+    alpha: Alpha,
 }
 
 impl Sdk {
-    /// Starts a new SDK instance, and connects to localhost on port 9357.
-    /// Blocks until connection and handshake are made.
-    /// Times out after ~30 seconds.
-    pub fn new() -> Result<Sdk> {
-        let port = env::var("AGONES_SDK_GRPC_PORT").unwrap_or("9357".to_string());
-        let addr = format!("localhost:{}", port);
-        let env = Arc::new(grpcio::EnvBuilder::new().build());
-        let ch = grpcio::ChannelBuilder::new(env)
-            .keepalive_timeout(Duration::new(30, 0))
-            .connect(&addr);
-        let cli = sdk_grpc::SdkClient::new(ch.clone());
-        let alpha = Alpha::new(ch);
-        let req = sdk::Empty::new();
+    /// Starts a new SDK instance, and connects to localhost on the port specified
+    /// or else falls back to the `AGONES_SDK_GRPC_PORT` environment variable,
+    /// or defaults to 9357.
+    ///
+    /// Note that this function will loop indefinitely until a connection is made
+    /// to the SDK server, so it is recommended to wrap this in a
+    /// [timeout](https://docs.rs/tokio/1.6.0/tokio/time/fn.timeout.html).
+    pub async fn new(port: Option<u16>, keep_alive: Option<Duration>) -> Result<Self> {
+        // TODO: Add TLS? For some reason the original Cargo.toml was enabling
+        // grpcio's openssl features, but AFAICT the SDK and sidecar only ever
+        // communicate via a non-TLS connection, so seems like we could just
+        // use the simpler client setup code if TLS is absolutely never needed
+        let addr: http::Uri = format!(
+            "http://localhost:{}",
+            port.unwrap_or_else(|| {
+                env::var("AGONES_SDK_GRPC_PORT")
+                    .ok()
+                    .and_then(|s| s.parse().ok())
+                    .unwrap_or(9357)
+            })
+        )
+        .parse()?;
+
+        let builder = tonic::transport::channel::Channel::builder(addr)
+            .keep_alive_timeout(keep_alive.unwrap_or_else(|| Duration::from_secs(30)));
+
+        // let mut root_store = rustls::RootCertStore::empty();
+        // root_store.add_server_trust_anchors(&webpki_roots::TLS_SERVER_ROOTS);
+        // let mut rusttls_config = rustls::ClientConfig::new();
+        // rusttls_config.alpn_protocols = vec![b"h2".to_vec(), b"http/1.1".to_vec()];
+        // rusttls_config.root_store = root_store;
+        // let tls_config =
+        //     tonic::transport::ClientTlsConfig::new().rustls_client_config(rusttls_config);
+        // builder = builder.tls_config(tls_config)?;
```

Comment:
This is just some commented out code for adding TLS, but it seems like this would never be used so I can just delete it.